test/python/test_localjob.py

```diff
@@ -79,10 +81,28 @@ def test_run_async(self):
         quantum_job = QuantumJob(qobj, backend, shots=1e5, preformatted=True)
```

Comment:
QuantumJob doesn't currently use shots when `preformatted=True`. Instead this could be,
```
qobj = qiskit._compiler.compile(qc, backend, compile_config={'shots':1e3})
quantum_job = QuantumJob(qobj, backend, preformatted=True)
```
This should be done in other tests here as well. Also, perhaps try increasing `num_qubits` to 10 or so to lengthen the simulation.